src/service/video-manager-impl.js

```diff
@@ -1598,17 +1620,33 @@ class VideoEntry {
    */
   finishDocking_() {
     // Remove draggable mask and listeners
-    this.finishDragging_();
-    // Re-enable controls
-    this.video.showControls();
-    // Restore the video inline
-    this.internalElement_.classList.remove(DOCK_CLASS);
-    this.internalElement_.setAttribute('style', '');
-    st.setStyles(dev().assertElement(this.video.element), {
-      'background-color': 'transparent',
+    this.vsync_.mutate(() => {
+      this.unlistenAll_();
+      this.hideControls(true, false);
+      this.showControls(false);
+      this.customControls_.toggleMinimalControls(false);
+      // Restore the video inline
+      this.video.element.classList.remove(DOCK_CLASS);
+      st.resetStyles(dev().assertElement(this.internalElement_),
```

Comment:
I'd advise against using that pattern, since it's creating a lot of unneeded closures. Instead, hoist helpers.